mobile-app/lib/ui/views/learn/widgets/quiz_widget.dart

```diff
@@ -21,54 +21,93 @@ class QuizWidgetQuestion {
   });
 }
 
-class QuizWidget extends StatelessWidget {
+class QuizWidget extends StatefulWidget {
   final List<QuizWidgetQuestion> questions;
   final Function(int, int) onChanged;
   final bool? isValidated;
 
-  const QuizWidget(
-      {super.key,
-      required this.questions,
-      required this.onChanged,
-      this.isValidated});
+  const QuizWidget({
+    super.key,
+    required this.questions,
+    required this.onChanged,
+    this.isValidated,
+  });
+
+  @override
+  State<QuizWidget> createState() => _QuizWidgetState();
+}
+
+class _QuizWidgetState extends State<QuizWidget> {
+  late final HTMLParser parser;
+  late List<List<Widget>> parsedQuestions;
+  late List<List<List<Widget>>> parsedOptions;
+
+  @override
+  void initState() {
+    super.initState();
+    parser = HTMLParser(context: context);
+    _parseAll();
+  }
+
+  @override
+  void didUpdateWidget(covariant QuizWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.questions != widget.questions) {
+      _parseAll();
+    }
+  }
+
+  void _parseAll() {
+    parsedQuestions =
+        widget.questions.map((q) => parser.parse(q.text)).toList();
+    parsedOptions = widget.questions
+        .map((q) => q.answers
+            .map((a) => parser.parse(
+                  a.answer,
+                  isSelectable: false,
+                  removeParagraphMargin: true,
+                ))
+            .toList())
+        .toList();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return Column(
-      children: List.generate(
-        questions.length,
-        (index) {
-          return quizQuestion(
-              context: context,
-              questionNumber: questions.length > 1 ? index + 1 : null,
-              question: questions[index],
-              selectedAnswer: questions[index].selectedAnswer,
-              isCorrect: questions[index].isCorrect,
-              onChanged: (answerIndex) {
-                onChanged(index, answerIndex);
-              });
-        },
-      ),
+    return ListView.builder(
+      shrinkWrap: true,
+      physics: const NeverScrollableScrollPhysics(),
+      itemCount: widget.questions.length,
+      itemBuilder: (context, index) {
+        return quizQuestion(
+          context: context,
+          questionNumber: widget.questions.length > 1 ? index + 1 : null,
+          question: widget.questions[index],
+          selectedAnswer: widget.questions[index].selectedAnswer,
+          isCorrect: widget.questions[index].isCorrect,
```

Comment:
Since we're passing `questionIndex` why not get the question, `selectedAnswer` and `isCorrect` using that and `widget.questions` in the respective widget its required. This way we don't need to pass another parameter.